libtenzir/builtins/contexts/geoip.cpp

```diff
@@ -378,14 +382,13 @@ class ctx final : public virtual context {
                            "geoip context can not be updated with events");
   }
 
-  auto update(chunk_ptr, context::parameter_map)
-    -> caf::expected<update_result> override {
-    return caf::make_error(ec::unimplemented, "geoip context can not be "
-                                              "updated with bytes");
+  auto make_query() -> caf::expected<make_query_type> override {
+    return caf::make_error(ec::unimplemented,
+                           "geoip context doesn't support retroactive lookups");
```

Comment:
```suggestion
    return {};
```

This error gets swallowed by the post-load query expression emittance loop. It will never be displayed.

Contexts that do not support retroactive lookups should return an empty function.

If anything, this error should appear when you want to _only_ create a retrospective lookup for a context that doesn't support them, but that's not in the scope of this PR.